Source/Modules/php.cxx

```diff
@@ -1270,11 +1298,23 @@ class PHP : public Language {
     else if (constructor) {
       wname = NewString("__construct");
     }
-    else if (wrapperType == membervar || wrapperType == globalvar) {
+    else if (wrapperType == membervar) {
       char *ptr = Char(iname);
       ptr+= strlen(Char(iname)) - 4 - strlen(Char(name));
       wname = (String*) ptr;
     }
+    else if (wrapperType == globalvar) {
+      //check for namespaces
+      String *nameSpace = getNameSpace(GetChar(n, "name"));
+      if (nameSpace == NULL) {
+        char *ptr = Char(iname);
+        ptr+= strlen(Char(iname)) - 4 - strlen(Char(name));
+        wname = (String*) ptr;
+      }
+      else {
+        wname = iname;
+      }
+    }
     else if (wrapperType == staticmembervar) {
       // Shape::nshapes -> nshapes
       char *ptr = Char(strrchr(GetChar(n, "name"),':'));
```

Comment:
`Char(x)` is a SWIG thing to get you a `char *` pointer for a `DOH` type.

But `strrchr()` is a standard C function which returns a `char*`, so at best applying `Char()` to the returned value is useless here.

However, I don't think you should be parsing the C++ type name here, as that will ignore any `%rename` in effect.  I think you probably want something like `char * ptr = Char(Getattr(n, "staticmembervariableHandler:sym:name"));` here.